src/app/component/MissionDetailsCard.jsx

```diff
@@ -0,0 +1,384 @@
+              <Avatar className={classes.avatar} alt="Volunteer" src={avatarImage} />
+            ) : null}
+          </Grid>
+          <Grid item xs={11}>
+            {content}
+          </Grid>
+        </React.Fragment>
+      );
+    })}
+  </Grid>
+);
+
+const MissionDetailsContent = ({ description, classes }) => (
+  <Box>
+    <H5 align="left" color="textSecondary">
+      Mission Type
+    </H5>
+    <Body2 align="left" className={classes.missionTypeText} color="textPrimary">
+      {description}
+    </Body2>
+  </Box>
+);
+
+const MissionDetailsButton = ({
+  type,
+  volunteerForMission,
+  startMission,
+  markMissionAsDelivered,
+}) => {
+  switch (type) {
+    case MissionStatus["unassigned"]:
+      return (
+        <StyledButton
+          color="primary"
+          variant="contained"
+          disableElevation
+          onClick={volunteerForMission}
+        >
+          Accept Mission
+        </StyledButton>
+      );
+    case MissionStatus["tentative"]:
+    case MissionStatus["assigned"]:
+      return (
+        <StyledButton color="primary" variant="contained" disableElevation onClick={startMission}>
+          Start Mission
+        </StyledButton>
+      );
+    case MissionStatus["started"]:
+      return (
+        <StyledButton
+          color="primary"
+          variant="contained"
+          disableElevation
+          onClick={markMissionAsDelivered}
+        >
+          Mark Mission as Delivered
+        </StyledButton>
+      );
+    case MissionStatus["delivered"]:
+    case MissionStatus["done"]:
+      return null;
+    default:
+      return (
+        <StyledButton color="primary" variant="contained" disableElevation>
+          Loading..
+        </StyledButton>
+      );
+  }
+};
+
+const MissionDetailsPickUpDeliveryHeader = ({ header, classes }) => (
+  <Body2 align="left" className={classes.deliveryDetailsHeader} color="textPrimary">
+    {header}
+  </Body2>
+);
+
+/**
+ * Component for displaying mission details as a card
+ *
+ * @component
+ */
+const MissionDetailsCard = ({
+  showBackIcon,
+  mission,
+  volunteer,
+  volunteerForMission,
+  startMission,
+  markMissionAsDelivered,
+  unassignFromMission,
+  userUnverifiedPopupOpen,
+  setUserUnverifiedPopupOpen,
+  history,
+}) => {
+  const classes = useStyles();
+  showBackIcon = showBackIcon || false;
+  const title = mission.title || null;
+  const status = mission.status || null;
+  const fundedStatus = mission.fundedStatus
+    ? titleCase(MissionFundedStatus[mission.fundedStatus])
+    : null;
+  const description = mission.description || null;
+  const pickUpLocation = mission.pickUplocation || null;
+  const pickUpWindow = mission.pickUpWindow || null;
+  const deliveryLocation = mission.deliverylocation || null;
+  const deliveryWindow = mission.deliveryWindow || null;
+  const recipientName = mission.recipientName || null;
+  const recipientPhoneNumber = mission.recipientPhoneNumber || null;
```

Comment:
Maybe look at PropTypes.defaultProps to set all these defaults?

```
MissionDetailsCard.defaultProps = { 
  // default goes here
};
```

Reply:
Didn't know about PropTypes.defaultProps until now, thanks for pointing that out.